tests.py

```diff
@@ -358,6 +366,52 @@ class TestBinaryModulatedWriter(TestWriter):
     label = "TestBinaryModulatedWriter"
     writer_class_to_test = pyfstat.BinaryModulatedWriter
 
+    def test_tp_parsing(self):
+        this_writer = self.writer_class_to_test(
+            outdir=self.outdir, **default_Writer_params, **default_binary_params,
+        )
+        this_writer.make_data()
+
+        theta_prior = {
+            key: value
+            for key, value in default_signal_params.items()
+            if key not in ["h0", "cosi"]
+        }
+        theta_prior.update({key: value for key, value in default_binary_params.items()})
+        theta_prior["tp"] = {
+            "type": "unif",
+            "lower": default_binary_params["tp"]
+            - 0.5 * default_binary_params["period"],
+            "upper": default_binary_params["tp"]
+            + 0.5 * default_binary_params["period"],
+        }
+        theta_prior.pop
```

Comment:
Why is the `pop` needed? (not a request for change, just want to make sure I'm not missing something similar in other use cases)

Reply:
Because `default_signal_params` includes `h0` and `cosi`, and I'm using a comprehension to fill up the `theta_prior`. I can manually fill up `theta_prior` if you prefer it.

Reply:
Ok if it works!